server/app/controllers/admin/AdminProgramBlockPredicatesController.java

```diff
@@ -426,4 +431,14 @@ public Result updateEligibilityMessage(Request request, long programId, long blo
     return redirect(indexUrl)
         .flashing(toastType.toString().toLowerCase(Locale.getDefault()), toastMessage);
   }
+
+  /** HTMX partial that renders a card for editing a condition within a predicate. */
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result hxEditCondition(Request request) {
+    DynamicForm formData = formFactory.form().bindFromRequest(request);
+    long conditionId = Long.parseLong(formData.get("conditionId"));
+    return ok(editConditionPartialView.render(
+            request, new EditConditionPartialViewModel(conditionId)))
+        .as(Http.MimeTypes.HTML);
+  }
 }
```

Comment:
(blocking) This doesn't handle errors like an empty request, validation, parsing, or rendering issues. If this fails how should the UI react for the user? Since it's being called by HTMX this should ideally never throw so we can show something on the screen.

I recommend forgoing the use of DynamicForm and creating a typed object that the formFactory can bind to, then you can leverage the build in ability to do validation. See [here](https://github.com/civiform/civiform/blob/cc45bb74df14b29eef232b6e9078abd5f4a1ba46/server/app/controllers/admin/apibridge/ProgramBridgeController.java#L151-L158) and [here](https://github.com/civiform/civiform/blob/main/server/app/views/admin/apibridge/programbridge/ProgramBridgeEditCommand.java#L11).